specification/flag-evaluation/hooks.md

```diff
@@ -2,4 +2,58 @@
 
 ## Overview
 
-// TODO: this is a stub.
+Hooks are a mechanism which can tie into the lifecycle of flag evaluation. They
+operate similarly to middleware in many web frameworks.
+
```

Comment:
"tie in" is vague. I suggest something like "hooks are a mechanism by which arbitrary behavior (or maybe "logic" if you prefer) to flag evaluation".